client/state/document-head/reducer.js

```diff
@@ -47,17 +47,19 @@ export const meta = withSchemaValidation( metaSchema, ( state = DEFAULT_META_STA
 
 export const link = withSchemaValidation( linkSchema, ( state = [], action ) => {
 	switch ( action.type ) {
-		case DOCUMENT_HEAD_LINK_SET:
+		case DOCUMENT_HEAD_LINK_SET: {
 			if ( ! action.link ) {
 				return state;
 			}
 
 			// Append action.link to the state array and prevent duplicate objects.
 			// Works with action.link being a single link object or an array of link objects.
-			return uniqWith(
-				[ ...state, ...( Array.isArray( action.link ) ? action.link : [ action.link ] ) ],
-				isEqual
-			);
+			const links = Array.isArray( action.link ) ? action.link : [ action.link ];
+			state = links.reduce( ( accuState, newLink ) => {
```

Comment:
Is it intended to not return the result of `links.reduce` but assign it to `state`?

Reply:
Oh yes, it should `return`. I'll need to create a unit test for this, apparently it didn't get caught.

Reply:
@flootr I just fixed this bug in 65421c7f45 and also pushed a few commits with tests that verify the deduplication logic. There's one `chai` -> `jest` migration, too. Ready for final review.